tests/test_func.py

```diff
@@ -36,57 +37,60 @@
 INFO_TEST_RGX = re.compile(r"^func_i\d\d\d\d$")
 
 
-def exception_str(self, ex):  # pylint: disable=unused-argument
-    """function used to replace default __str__ method of exception instances"""
+def exception_str(self, ex) -> str:  # pylint: disable=unused-argument
+    """function used to replace default __str__ method of exception instances
+    This function is not typed because it is legacy code"""
     return f"in {ex.file}\n:: {', '.join(ex.args)}"
 
 
 class LintTestUsingModule:
-    INPUT_DIR = None
+    INPUT_DIR: Optional[str] = None
     DEFAULT_PACKAGE = "input"
     package = DEFAULT_PACKAGE
     linter = linter
-    module = None
-    depends = None
-    output = None
+    module: Optional[str] = None
+    depends: Optional[List[Tuple[str, str]]] = None
+    output: Optional[str] = None
 
-    def _test_functionality(self):
-        tocheck = [self.package + "." + self.module]
+    def _test_functionality(self) -> None:
+        if self.module:
+            tocheck = [self.package + "." + self.module]
         # pylint: disable=not-an-iterable; can't handle boolean checks for now
         if self.depends:
             tocheck += [
                 self.package + f".{name.replace('.py', '')}" for name, _ in self.depends
             ]
         self._test(tocheck)
 
-    def _check_result(self, got):
+    def _check_result(self, got: str) -> None:
         error_msg = (
             f"Wrong output for '{self.output}':\n"
             "You can update the expected output automatically with: '"
             f"python tests/test_func.py {UPDATE_OPTION}'\n\n"
         )
         assert self._get_expected() == got, error_msg
 
-    def _test(self, tocheck):
-        if INFO_TEST_RGX.match(self.module):
+    def _test(self, tocheck: List[str]) -> None:
+        if self.module and INFO_TEST_RGX.match(self.module):
             self.linter.enable("I")
         else:
             self.linter.disable("I")
         try:
             self.linter.check(tocheck)
         except Exception as ex:
-            # need finalization to restore a correct state
-            self.linter.reporter.finalize()
-            ex.file = tocheck
+            print(f"Exception: {ex} in {tocheck}:: {'‚ '.join(ex.args)}")
+            ex.file = tocheck  # type: ignore # This is legacy code we're trying to remove, impossible to type correctly
             print(ex)
-            ex.__str__ = exception_str
+            ex.__str__ = exception_str  # type: ignore # This is legacy code we're trying to remove, impossible to type correctly
             raise
         self._check_result(self.linter.reporter.finalize())
 
-    def _has_output(self):
-        return not self.module.startswith("func_noerror_")
+    def _has_output(self) -> bool:
+        if self.module and not self.module.startswith("func_noerror_"):
+            return True
+        return False
```

Comment:
```suggestion
        return self.module and not self.module.startswith("func_noerror_")
```

Reply:
I did this because `mypy` fails to understand that because of `self.module` at the beginning of L89 `self.module` can't be None. I can add a `type: ignore` but thought it would be more elegant to show that the return value will always be `bool`.

Reply:
Maybe ``bool(self.module)`` then ?

Reply:
Is used `isinstance()`. Going through tests now!

Reply:
This works!